sharness/t0042-basic-auth.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+test_description="Test service + ctl SSL interaction"
+
+config="`pwd`/config/basic_auth"
+
+. lib/test-lib.sh
+
+test_ipfs_init
+cleanup test_clean_ipfs
+test_cluster_init "$config"
+cleanup test_clean_cluster
+
+test_expect_success "prerequisites" '
+    test_have_prereq IPFS && test_have_prereq CLUSTER
+'
+
+test_expect_success "BasicAuth fails without credentials" '
+    id=`cluster_id`
+    ipfs-cluster-ctl id | grep -q Unauthorized
```

Comment:
I'm not convinced checking for 'Unauthorized' in the response is the best approach to this -- e.g. we could accidentally be leaking info in some case but the presence of the word 'Unauthorized' would allow this test to pass anyway.

Reply:
@hsanjuan

Reply:
@dgrisham check that the command did not return 0 for exit code and then check for Unauthorized (should be ok for the moment).